Content.Client/Silicons/Laws/Ui/LawDisplay.xaml.cs

```diff
@@ -38,51 +37,44 @@ public LawDisplay(EntityUid uid, SiliconLaw law, HashSet<string>? radioChannels)
         if (!_entityManager.TryGetComponent<SpeechComponent>(uid, out var speech) || speech.SpeechSounds is null)
             return;
 
-        var localButton = new Button
+        // New unified button to work with drop down list
+        var stateLawButton = new Button
         {
-            Text = Loc.GetString("hud-chatbox-select-channel-Local"),
-            Modulate = Color.DarkGray,
+            Text = Loc.GetString("laws-ui-state-law"),
+            Modulate = Color.LightBlue,
             StyleClasses = { "chatSelectorOptionButton" },
             MinHeight = 35,
-            MinWidth = 75,
+            MinWidth = 100
         };
 
-        localButton.OnPressed += _ =>
+        stateLawButton.OnPressed += _ =>
         {
-            _chatManager.SendMessage($"{lawIdentifierPlaintext}: {lawDescriptionPlaintext}", ChatSelectChannel.Local);
-        };
-
-        LawAnnouncementButtons.AddChild(localButton);
-
-        if (radioChannels == null)
-            return;
-
-        foreach (var radioChannel in radioChannels)
-        {
-            if (!_prototypeManager.TryIndex<RadioChannelPrototype>(radioChannel, out var radioChannelProto))
-                continue;
-
-            var radioChannelButton = new Button
+            // Local and Binary channels are added to the list outside of the Radio component, so we handle them uniquely
+            if (selectedChannel == "Local")
+                _chatManager.SendMessage($"{lawIdentifierPlaintext}: {lawDescriptionPlaintext}", ChatSelectChannel.Local);
+            else if (selectedChannel == "Binary")
```

Comment:
these shouldn't be hardcoded. at the very least, use ValidatePrototypeId<T> or whatever it's called on a const string.

Reply:
Fixed